jupyter_server/services/sessions/handlers.py

```diff
@@ -47,12 +48,18 @@ async def post(self):
         if model is None:
             raise web.HTTPError(400, "No JSON data provided")
 
-        if "notebook" in model and "path" in model["notebook"]:
-            self.log.warning("Sessions API changed, see updated swagger docs")
-            model["path"] = model["notebook"]["path"]
-            model["type"] = "notebook"
+        if "notebook" in model:
+            if "name" in model["notebook"]:
+                model["path"] = model["notebook"]["name"]
+                model["type"] = "notebook"
+            elif "path" in model["notebook"]:
+                self.log.warning("Sessions API changed, see updated swagger docs")
+                model["path"] = model["notebook"]["path"]
+                model["type"] = "notebook"
```

Comment:
I figured the warning message was _because_ a "notebook" stanza existed in the model and a change was made afterward (since it's not listed in the model today). FWIW, Lab is not including a `notebook` stanza today (using 3.4.3).
```suggestion
        if "notebook" in model:
            self.log.warning("Sessions API changed, see updated swagger docs")
            model["type"] = "notebook"
            if "name" in model["notebook"]:
                model["path"] = model["notebook"]["name"]
            elif "path" in model["notebook"]:
                model["path"] = model["notebook"]["path"]
```